stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/sql/BMirrorTable.java

```diff
@@ -0,0 +1,199 @@
+        } finally {
+            SQLDatasourceUtils.cleanupConnection(null, null, conn, isInTransaction);
+        }
+    }
+
+    public void performRemoveOperation(Context context, BFunctionPointer lambdaFunction) {
+        int deletedCount = 0;
+        Connection connection = null;
+        boolean isInTransaction = context.isInTransaction();
+        try {
+            connection = SQLDatasourceUtils
+                    .getDatabaseConnection(context, this.getDatasource(), isInTransaction);
+            if (!isInTransaction) {
+                connection.setAutoCommit(false);
+            }
+            while (this.hasNext(false)) {
+                BStruct data = this.getNext();
+                BValue[] args = { data };
+                BValue[] returns = BLangFunctions
+                        .invokeCallable(lambdaFunction.value().getFunctionInfo(),
+                                args);
+                if (((BBoolean) returns[0]).booleanValue()) {
+                    ++deletedCount;
+                    this.removeData(data, connection);
+                }
+            }
+            if (!isInTransaction) {
+                connection.commit();
+            }
+            context.setReturnValues(new BInteger(deletedCount));
+        } catch (SQLException e) {
+            context.setReturnValues(TableUtils.createTableOperationError(context, e));
+            SQLDatasourceUtils.handleErrorOnTransaction(context);
+            SQLDatasourceUtils.cleanupConnection(null, null, connection, isInTransaction);
+        }
+    }
+
+    public void removeData(BStruct data, Connection conn) throws SQLException {
+        PreparedStatement stmt = null;
+        try {
+            String sqlStmt = TableUtils.generateDeleteDataStatment(tableName, data);
+            stmt = conn.prepareStatement(sqlStmt);
+            TableUtils.prepareAndExecuteStatement(stmt, data);
+        }  finally {
+            if (stmt != null) {
+                stmt.close();
+            }
+        }
+    }
+
+    @Override
+    public void next() {
+        if (((SQLDataIterator) iterator).getResultset() == null) {
+            populateIteratorAttributes(tableName);
+            this.nextPrefetched = false;
+            this.hasNextVal = false;
+        }
+        if (!nextPrefetched) {
+            iterator.next();
+        } else {
+            nextPrefetched = false;
+        }
+    }
+
+    @Override
+    public boolean hasNext(boolean isInTransaction) {
+        if (((SQLDataIterator) iterator).getResultset() == null) {
+            populateIteratorAttributes(tableName);
+            this.nextPrefetched = false;
+            this.hasNextVal = false;
+        }
+        if (!nextPrefetched) {
+            hasNextVal = iterator.next();
+            nextPrefetched = true;
+        }
+        if (!hasNextVal) {
+            close(isInTransaction);
+        }
+        return hasNextVal;
```

Comment:
Keep the above logic in base class and just implement the BMirrorTable unique logic only here.